src/runtime_src/core/edge/sw_em/generic_pcie_hal2/halapi.cxx

```diff
@@ -71,14 +72,22 @@ xclDeviceHandle xclOpen(unsigned deviceIndex, const char *logfileName, xclVerbos
   else
   {
     handle = new xclcpuemhal2::CpuemShim(deviceIndex,info,DDRBankList,false,false,fRomHeader);
```

Comment:
use std::unique_ptr.  Use appropriate calls to .get() when passing raw pointer to functions like handleCheck, and use release() at line 91.  Remove delete on line 79

Reply:
All the necessary checks and deletions handled properly in required places. as you suggested will use smart pointers for this. we will handle this 2020.2.